ImperatorToCK3/Source/CK3/CK3World.cpp

```diff
@@ -0,0 +1,170 @@
+	// ---- Loading history/provinces
+	auto fileNames = Utils::GetAllFilesInFolder(ck3Path + "/game/history/provinces/");
+	for (const auto& fileName : fileNames)
+	{
+		if (fileName.find(".txt") == std::string::npos)
+			continue;
+		try
+		{
+			auto newProvinces = Provinces(ck3Path + "/game/history/provinces/" + fileName);
+			for (const auto& newProvince : newProvinces.getProvinces())
+			{
+				const auto id = newProvince.first;
+				if (provinces.count(id))
+				{
+					Log(LogLevel::Warning) << "Vanilla province duplication - " << id << " already loaded! Overwriting.";
+					provinces[id] = newProvince.second;
+				}
+				else
+					provinces.insert(std::pair(id, newProvince.second));
+			}
+		}
+		catch (std::exception& e)
+		{
+			Log(LogLevel::Warning) << "Invalid province filename: " << ck3Path << "/game/history/provinces/" << fileName << " : " << e.what();
+		}
+	}
+	
+	// now load the provinces that don't have unique entries in history/provinces
+	// they instead use history/province_mapping
+	fileNames = Utils::GetAllFilesInFolder(ck3Path + "/game/history/province_mapping/");
+	for (const auto& fileName : fileNames)
+	{
+		if (fileName.find(".txt") == std::string::npos)
+			continue;
+		try
+		{
+			auto newProvinces = ProvinceMappings(ck3Path + "/game/history/province_mapping/" + fileName);
+			for (const auto& newProvince : newProvinces.getMappings())
+			{
+				const auto id = newProvince.first;
+				if (provinces.find(newProvince.second) == provinces.end())
+				{
+					Log(LogLevel::Warning) << "Base province " << newProvince.second << " not found for province " << id << ".";
+					continue;
+				}
+				if (provinces.count(id))
+				{
+					Log(LogLevel::Info) << "Vanilla province duplication - " << id << " already loaded! Preferring unique entry over mapping.";
+				}
+				else
+					provinces.insert(std::pair(id, provinces.find(newProvince.second)->second));
+			}
+		}
+		catch (std::exception& e)
+		{
+			Log(LogLevel::Warning) << "Invalid province filename: " << ck3Path << "/game/history/province_mapping/" << fileName << " : " << e.what();
+		}
+	}
+	LOG(LogLevel::Info) << ">> Loaded " << provinces.size() << " province definitions.";
+}
+
+void CK3::World::importImperatorProvinces(const ImperatorWorld::World& sourceWorld)
+{
+	LOG(LogLevel::Info) << "-> Importing Imperator Provinces";
+	auto counter = 0;
+	// Imperator provinces map to a subset of CK3 provinces. We'll only rewrite those we are responsible for.
+	for (const auto& province : provinces)
+	{
+		const auto& impProvinces = provinceMapper.getImperatorProvinceNumbers(province.first);
+		// Provinces we're not affecting will not be in this list.
+		if (impProvinces.empty())
+			continue;
+		// Next, we find what province to use as its initializing source.
+		const auto& sourceProvince = determineProvinceSource(impProvinces, sourceWorld);
+		if (!sourceProvince)
+		{
+			continue; // MISMAP, or simply have mod provinces loaded we're not using.
```

Comment:
ck2toeu4 has this issue with mods - due to province_map.txt supporting provinces not originally in vanilla files (canada for sons of vinland), this would always scream when iterating over those. 
However, until there is such mod for ck3 you could restore the warning, and the author of vinland never actually came to us for support.